### PR TITLE
Switching from old "main" branch to new "abels" branch which

### PR DESCRIPTION
**Pull request by abelsalm**
I'm doing the PR because the project was started by William and then I worked on it, and now the main branch is outdated. 

All of the latest release were trained using the code from this branch.

I updated everything in "abels" so that it could be the new reference for this repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
+# Automatic segmentation of spinal canal
 
+<p align="center">
+  <img src="https://github.com/ivadomed/model-canal-seg/blob/abels/assets/canal_seg_visual.gif" alt="Segmentation on whole-spine image viewed on axial and sagittal planes" width="500">
+</p>
+
+This repository contains the code for deep learning-based segmentation of the spinal canal. 
+The code is based on the [nnUNet framework](https://github.com/MIC-DKFZ/nnUNet).
+
+The spinal canal as here been defined using the anatomical boundary of the dural sac. The model has been trained to segment both spinal cord and CSF inside the dural sac.
```
Comment (valosekj) [inline README.md:10]:
```suggestion
The spinal canal was defined using the anatomical boundary of the dural sac. The model was trained to segment all the structures within the dural sac, including the spinal cord, cerebrospinal fluid (CSF), and nerve rootlets.
```

File: README.md [hunk 2]
```diff
@@ -1,32 +1,41 @@
...
+</p>
+
+This repository contains the code for deep learning-based segmentation of the spinal canal. 
+The code is based on the [nnUNet framework](https://github.com/MIC-DKFZ/nnUNet).
+
+The spinal canal as here been defined using the anatomical boundary of the dural sac. The model has been trained to segment both spinal cord and CSF inside the dural sac. 
+
+## Model Overview
+
+The model is a 3d nnUNet which was trained specifically on T2-weighted images and provides and provides a segmentation of the spinal canal.
```
Comment (valosekj) [inline README.md:14]:
```suggestion
The model is a 3D nnUNet, which was trained on T2-weighted images to segment the spinal canal. 
```

File: README.md [hunk 3]
```diff
@@ -1,32 +1,41 @@
...
+
+## Model Overview
+
+The model is a 3d nnUNet which was trained specifically on T2-weighted images and provides and provides a segmentation of the spinal canal. 
+
+## How to use the model
+
+### Install dependencies
+
+- [Spinal Cord Toolbox (SCT) v6.5](https://github.com/spinalcordtoolbox/spinalcordtoolbox/releases/tag/6.2) or higher -- follow the installation instructions [here](https://github.com/spinalcordtoolbox/spinalcordtoolbox?tab=readme-ov-file#installation)
```
Comment (valosekj) [inline README.md:20]:
```suggestion
- [Spinal Cord Toolbox (SCT) v6.5](https://github.com/spinalcordtoolbox/spinalcordtoolbox/releases/tag/6.5) or higher -- follow the installation instructions [here](https://github.com/spinalcordtoolbox/spinalcordtoolbox?tab=readme-ov-file#installation)
```

File: postprocessing.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import numpy as np
```
Comment (valosekj) [inline postprocessing.py:1]: Please add a description. Otherwise it's not clear what the script is used to.

File: postprocessing.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            if not overwrite and os.path.exists(os.path.join(output_folder, file_name)):
+                output_file = os.path.join(output_folder, file_name.replace('.nii.gz', '_cleaned.nii.gz'))
+                process_segmentation_file(input_file, output_file)
+            else:
+                output_file = os.path.join(output_folder, file_name)
+                process_segmentation_file(input_file, output_file)
+
+
+input = "C:/Users/abels/OneDrive/Documents/NeuroPoly/canal_seg/segmentation/training/data/datasets/Dataset011_clean/labelsTr2"
+output = "C:/Users/abels/OneDrive/Documents/NeuroPoly/canal_seg/segmentation/training/data/datasets/Dataset011_clean/labelsTr2"
```
Comment (valosekj) [inline postprocessing.py:93]: I would replace the hardcoded paths by input args (using `argparser`)

File: preprocessing.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+            if file_path.endswith('.nii.gz'):
+                reorient(file_path)
+            # calls change_orientation on each file   
+    t2 = time.time()
+    print(f"Reorientation of {directory} done in {t2-t1} seconds")  
+
+# here I applied it to imagesTs, labelsTr and imagesTr
+apply_reorient_to_files("C:/Users/abels/OneDrive/Documents/NeuroPoly/canal_seg/segmentation/training/data/datasets/Dataset011_clean/imagesTs")
+apply_reorient_to_files("C:/Users/abels/OneDrive/Documents/NeuroPoly/canal_seg/segmentation/training/data/datasets/Dataset011_clean/labelsTr")
+apply_reorient_to_files("C:/Users/abels/OneDrive/Documents/NeuroPoly/canal_seg/segmentation/training/data/datasets/Dataset011_clean/imagesTr")
```
Comment (valosekj) [inline preprocessing.py:53]: the hardcoded paths are not ideal. try to think how to replace them, e.g., using input args.

File: preprocessing.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+            if chemin_cible is not None:
+                _transform_seg2image(chemin_base, chemin_cible, chemin_cible, override=True)
+                print(f"Header et affine remplacés pour : {nom_fichier_base} -> {nom_fichier_base.replace('_0000.nii.gz', '.nii.gz')}")
+            else:
+                print(f"Aucune correspondance trouvée pour {nom_fichier_base}")
+    t2 = time.time()
+    print(f"Registration done in {t2-t1} seconds")
+
+# apply to the training set
+register_seg_to_image("C:/Users/abels/OneDrive/Documents/NeuroPoly/canal_seg/segmentation/training/data/datasets/Dataset011_clean/imagesTr", "C:/Users/abels/OneDrive/Documents/NeuroPoly/canal_seg/segmentation/training/data/datasets/Dataset011_clean/labelsTr")
```
Comment (valosekj) [inline preprocessing.py:164]: idem

File: image.py [hunk 8]
```diff
@@ -0,0 +1,601 @@
+# flake8: noqa
```
Comment (valosekj) [inline image.py:1]: Add description. The script is probably copy-pasted from SCT? If so, mention this in the description

File: README.md (hunk 3 above)
Comment (abelsalm) [reply README.md:20]: Sure! I should definitively have not forgot that !

File: postprocessing.py (hunk 4 above)
Comment (abelsalm) [reply postprocessing.py]: Done !

**Review by valosekj — APPROVED** [review]

**Comment by valosekj** [comment]
Thanks for addressing my comments, @abelsalm. 👍🏻  Approving. You can merge.

**Pull request merged**